testing/test_collection.py

```diff
@@ -1207,26 +1207,25 @@ def test_1():
 def test_collect_pyargs_with_testpaths(
     pytester: Pytester, monkeypatch: MonkeyPatch
 ) -> None:
-    testmod = pytester.mkdir("testmod")
-    # NOTE: __init__.py is not collected since it does not match python_files.
-    testmod.joinpath("__init__.py").write_text("def test_func(): pass")
-    testmod.joinpath("test_file.py").write_text("def test_func(): pass")
+    for i in range(3):
+        testpkg = pytester.mkpydir(f"tests{i}")
+        testmod = pytester.mkdir(str(testpkg.joinpath("testmod")))
+        # NOTE: __init__.py is not collected since it does not match python_files.
+        testmod.joinpath("__init__.py").write_text("def test_func(): pass")
+        testmod.joinpath("test_file.py").write_text("def test_func(): pass")
 
-    root = pytester.mkdir("root")
-    root.joinpath("pytest.ini").write_text(
-        textwrap.dedent(
-            """
+    pytester.makeini(
+        """
         [pytest]
         addopts = --pyargs
```

Comment:
This test seems like the wrong place to test the globbing functionality, as it specificially tests behavior with the `--pyargs` flag (see #4405 / a252e81ced6f077e3ee880bed7b7147962c79c34).

Instead, this should probably be tested in `test_testpaths_ini`, or, perhaps cleaner, in an entirely separate test.